docs/capacity-planning.rst

```diff
@@ -0,0 +1,326 @@
+        
+          Sub-topology: 1
+            Source: KSTREAM-SOURCE-0000000010 (topics: [KSTREAM-AGGREGATE-STATE-STORE-0000000006-repartition])
+              --> KSTREAM-AGGREGATE-0000000007
+            Processor: KSTREAM-AGGREGATE-0000000007 (stores: [KSTREAM-AGGREGATE-STATE-STORE-0000000006])
+              --> KTABLE-MAPVALUES-0000000011
+              <-- KSTREAM-SOURCE-0000000010
+            Processor: KTABLE-MAPVALUES-0000000011 (stores: [])
+              --> KTABLE-TOSTREAM-0000000012
+              <-- KSTREAM-AGGREGATE-0000000007
+            Processor: KTABLE-TOSTREAM-0000000012 (stores: [])
+              --> KSTREAM-MAPVALUES-0000000013
+              <-- KTABLE-MAPVALUES-0000000011
+            Processor: KSTREAM-MAPVALUES-0000000013 (stores: [])
+              --> KSTREAM-SINK-0000000014
+              <-- KTABLE-TOSTREAM-0000000012
+            Sink: KSTREAM-SINK-0000000014 (topic: PAGEVIEWS_BY_PAGE)
+              <-- KSTREAM-MAPVALUES-0000000013
+
+    Observe that there are 2 sub-topologies. This means that the stream is being re-partitioned.
+
+**State Store Changelog Topics**
+
+KSQL uses an embedded storage engine to manage state locally for operations such as aggregations. For fault-tolerance reasons it also persists the state for aggregations (e.g, SUM, COUNT, TOPKDISTINCT) in a compacted changelog topic. The changelog topic has the same number of partitions as the input stream. It defaults to a single replica, but this can be explicitly set via the ``ksql.streams.replication.factor`` property.
+
+The amount of data stored in the changelog topic depends on the number of keys, key size, aggregate size, and whether the aggregation is windowed and if so, what the window retention time is. 
+
+For un-windowed aggregations the total size should be roughly the (key size + aggregate size) multiplied by the number of keys.
+
+For windowed aggregates the size is determined by the number of outstanding windows multiplied by the size of each window. The number of outstanding windows  is bound by the window retention time. The size of each window depends on message throughput, key space size and the average key size. If you have a large key space then each window’s size will likely be a multiple of the throughput, window size, and average key size. If your key space is small then the window’s size is bound by the number of keys multiplied by the average key size.
+
+**KSQL Command Topics**
+
+When run in interactive configuration, a KSQL cluster creates an internal topic (whose name is derived from the ksql.service.id setting) to persist the log of queries to run across all the servers in this KSQL cluster. These special-purpose topics for KSQL are called command topics.  Command topics have a single partition and default to a replication factor of 1.
+
+Consumption/Production
+++++++++++++++++++++++
+
+You may need to provision additional Kafka brokers to accommodate KSQL production/consumption to/from your Kafka cluster.
```

Comment:
suggest: `You might need to provision additional Kafka brokers to accommodate KSQL production and consumption to and from your Kafka cluster.`